src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -136,6 +140,52 @@ public void Dispose()
         public DataFrame Table(string tableName) =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("table", tableName));
 
+        /// <summary>
+        /// Returns a dataframe as per the schema and data.
+        /// </summary>
+        /// <param name="data">List of Row objects</param>
+        /// <param name="schema">Schema as StructType</param>
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data, StructType schema) =>
+            new DataFrame((JvmObjectReference)_jvmObject.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.sql.api.dotnet.SQLUtils",
+                "createDataFrameHelper",
+                this,
+                data,      
+                DataType.FromJson(_jvmObject.Jvm, schema.Json)));
+
+        /// <summary>
+        /// Returns a dataframe when given only the data and no schema
+        /// </summary>
+        /// <param name="data">List of Row objects</param>        
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data)
+        {                       
+            var schemaFields = new List<StructField>();                        
+            object[] values = data.First().Values;
+            for (int i = 0; i < values.Length; ++i)
+            {
+                string valueType = values[i].GetType().Name;
+                StructField field;
+                switch (valueType)
```

Comment:
I was thinking more in the lines of 
https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#using-pattern-matching-switch-statements

Reply:
Made the pattern matching change, please let me know if this is what you had in mind. Thanks!